extensions/wikia/ExactTargetUpdates/tests/ExactTargetAddUserTaskTest.php

```diff
@@ -1,76 +1,59 @@
 <?php
 
-include __DIR__ . '/../lib/exacttarget_soap_client.php';
+require_once __DIR__ . '/../lib/exacttarget_soap_client.php';
 
 class ExactTargetAddUserTaskTest extends WikiaBaseTest {
 
-
-	function testSendNewUserShouldInvokeCreateSubscriber() {
+	function testSendNewUserShouldDistributeParams() {
 		/* Params to compare */
 		$aUserData = [
 			'user_id' => '12345',
 			'user_email' => 'email@email.com'
 		];
 		$aUserProperties = [
-			'property1' => 'value1',
-			'property2' => 'value2'
+			'property_name' => 'property_value'
 		];
 
-		$addTaskMock = $this->getMockBuilder( 'ExactTargetAddUserTask' )
+		$oSoapClient = $this->getMockBuilder( 'ExactTargetSoapClient' )
 			->disableOriginalConstructor()
-			->setMethods( [ 'createUserPropertiesDataExtension', 'createUserDataExtension', 'createSubscriber' ] )
+			->setMethods( [ 'Create' ] )
 			->getMock();
-		$addTaskMock
-			->expects( $this->once() )
-			->method( 'createSubscriber' )
-			->will(  $this->returnValue( NULL ) );
-
-		$addTaskMock->sendNewUserData( $aUserData, $aUserProperties );
-	}
-
-	function testSendNewUserShouldInvokecreateUserDataExtension() {
-		/* Params to compare */
-		$aUserData = [
-			'user_id' => '12345',
-			'user_email' => 'email@email.com'
-		];
 
+		/* Mock tested class /*
+		/* @var ExactTargetAddUserTask $addTaskMock mock of ExactTargetAddUserTask class */
 		$addTaskMock = $this->getMockBuilder( 'ExactTargetAddUserTask' )
 			->disableOriginalConstructor()
-			->setMethods( [ 'createUserPropertiesDataExtension', 'createUserDataExtension', 'createSubscriber' ] )
+			->setMethods( [ 'getClient', 'createUserPropertiesDataExtension', 'createUserDataExtension', 'createSubscriber' ] )
 			->getMock();
+
 		$addTaskMock
 			->expects( $this->once() )
-			->method( 'createUserDataExtension' )
-			->will(  $this->returnValue( NULL ) );
+			->method( 'getClient' )
```

Comment:
Tests should be invoked against the objects that you are trying to test and not the mocks. If you find yourself having to test the mocks and override dependent methods it's often an indication that you need to add some form of dependency injection.

For example, in this case the object being tested should be `ExactTargetAddUserTask` and not the mock. Instead of "mocking" `getClient` there should be a way to inject the client (`setClient($client)`) so you can mock it's behavior.